enhancements/installer/aws-custom-dns.md

```diff
@@ -0,0 +1,475 @@
+solution [Route53 for AWS](https://aws.amazon.com/route53/). OpenShift will
+continue to configure and manage the cloud load balancing solution [Elastic
+Load Balancer](https://aws.amazon.com/elasticloadbalancing/) for all its
+services.
+
+OpenShift will start CoreDNS pods to provide DNS resolution for Internal API
+and Ingress services that are essential for cluster creation.
+
+### Workflow Description
+
+#### API Server Workflow
+
+For an OpenShift administrator to use a custom DNS solution rather than
+letting OpenShift automatically configure the cloud provided DNS solution,
+they would have to do the following things:
+- Let the OpenShift Installer know that they would like to use a custom
+DNS solution and skip configuration of the cloud provided DNS solution.
+
+- Configure the custom DNS solution with a record for the API Server. The
+`Design Details` section goes into further detail regarding this.
+
+The Installer does not generate terraform configuration for the cloud's
+cloud's DNS solution(Route53 for AWS). It continues to generate terraform
+Load Balancer configuration for the API Server.
+
+#### Internal API Server Worflow
+
+The Installer instructs the machine-config-operator(MCO) to start the CoreDNS
+pod on the bootstrap and master nodes to provide DNS resolution for the
+Internal API service. This is a pattern already used in on-prem IPI
+installations. This implementation differs from the on-prem case by not
+staring the keepalived or haproxy pods for load balancing.
+
+When private subnets are created by the admin before install, the Installer
+passes the IP addresses for NICs on the private subnets to the MCO via the
+AWSPlatformStatus within the Infrastructure CR. These IP addresses are provided
+to the template plugin via the CoreDNS Corefile.
+
+What happens when private subnets are not in use? Can the user be expected to
+provide the IP addresses within the machine network for API-Int Server?
+
+Using the same IP addresses as the API server is also an option but would
+not provide the benefit of a [Split-Horizon DNS](https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/hosted-zone-private-considerations.html#hosted-zone-private-considerations-split-view-dns).
+
+#### Ingress Service Workflow
+
+The Installer omits the DNS Zones within `dnses.config.openshift.io/cluster`.
+This influences the behavior of the Ingress Operator by making it skip the
+configuration of cloud DNS while continuing with Load Balancer management.
+
+The CoreDNS pod managed by the MCO is configured to also provide DNS resolution
+for the Ingress Service since this service is essential for cluster bringup. To
+support all IngressControllers and not just the default one, MCO needs to look 
+up all services in the `openshift-ingress` namespace. Then select the services 
+of `type: LoadBalancer` and with a label having the key 
+`ingresscontroller.operator.openshift.io/owning-ingresscontroller`. These
```

Comment:
It is different from baremetal networking because in this case, the Ingress controller is responsible for configuring the LoadBalancer whereas in the baremetal case the CoreDNS pod handled that too. With regards to just the DNS part, I have tried to keep the design close to the baremetal case.